test/src/test/java/org/corfudb/runtime/checkpoint/CheckpointTrimTest.java

```diff
@@ -210,21 +210,76 @@ public void rawStreamConsumer() {
                 .setStreamName(tableName)
                 .open();
 
-        IStreamView s = newRuntime.getStreamsView().get(CorfuRuntime.getStreamID(tableName));
-        s.seek(BATCH_SIZE + CHECKPOINT_SIZE);
-        // Seek beyond the last trimmed address.
+        // Create a new stream and seek beyond the last trimmed address.
+        IStreamView stream = newRuntime.getStreamsView().get(CorfuRuntime.getStreamID(tableName));
+        stream.seek(BATCH_SIZE + CHECKPOINT_SIZE);
+
         // The first call to remainingUpTo() will load the checkpoint, and the
         // second one will fetch the actual data.
-        Assertions.assertThat(Stream.of(s.remainingUpTo(Long.MAX_VALUE), s.remainingUpTo(Long.MAX_VALUE))
+        Assertions.assertThat(Stream.of(
+                stream.remainingUpTo(Long.MAX_VALUE), stream.remainingUpTo(Long.MAX_VALUE))
                 .map(List::size).mapToInt(Integer::intValue).sum())
                 .isEqualTo(BATCH_SIZE);
 
         trim(mcw.appendCheckpoints(getRuntime(), CHECKPOINT_AUTHOR));
         IntStream.range(0, BATCH_SIZE).forEach(idx -> newMap.put(idx, idx));
-        Assertions.assertThatThrownBy(() -> s.remainingUpTo(Long.MAX_VALUE))
+        Assertions.assertThatThrownBy(() -> stream.remainingUpTo(Long.MAX_VALUE))
```

Comment:
Good catch! Let me look into this.

Reply:
@annym Can you take a look at the current approach? Basically we force holes to end up in the resolved queue, which prevents TrimmedException.

Reply:
Can we not fix this by not short-circuiting the flow in fillReadQueue(...), so that after we lead the checkpoint, we do not                     `return true`?

Reply:
We seem to have a notion of the maximum resolved address via the `globalPointer` within `AbstractStreamContext`. After making `globalPointer` hole-aware, the problem does not happen.

```
        // Update the global pointer
        if (readFrom.size() > 0) {
            context.setGlobalPointerCheckGCTrimMark(readFrom.get(readFrom.size() - 1)
                    .getGlobalAddress());
        }
```